### PR TITLE
[incubator-kie-issues-1309] Multiinstance loop

### PR DESCRIPTION
**Pull request by elguardian**
Closes: https://github.com/apache/incubator-kie-issues/issues/1309

### CODE REVIEW

**Comment by kie-ci3** [comment]
**PR job** `#2` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3554 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3554/2/display/redirect

**Test results:**
- PASSED: 3271
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3554/2/testReport/org.jbpm.bpmn2/FlowTest/testMultiInstanceLoopCharacteristicsTask/">org.jbpm.bpmn2.FlowTest.testMultiInstanceLoopCharacteristicsTask</a></summary>
Build Errors:<br/>Error Messages:<br/>Message [id=1, kieBase=defaultKieBase, level=ERROR, path=BPMN2-MultiInstanceLoopCharacteristicsTask.bpmn2, line=-1, column=0<br/>   text=Unable to load process. : Exception class java.io.FileNotFoundException : 'BPMN2-MultiInstanceLoopCharacteristicsTask.bpmn2' cannot be opened because it does not exist]<br/>---<br/>Warning Messages:<br/>---<br/>Info Messages:<br/>
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3554/2/testReport/org.jbpm.bpmn2/IntermediateEventTest/testMultiInstanceLoopBoundaryTimer/">org.jbpm.bpmn2.IntermediateEventTest.testMultiInstanceLoopBoundaryTimer</a></summary>
<br/>expected: 2<br/> but was: 1
</details>

File: jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/ForEachNodeInstance.java [hunk 1]
```diff
@@ -319,21 +317,13 @@ private boolean areNodeInstancesCompleted() {
...
-                        throw new IllegalArgumentException("Completion condition expression must return boolean values: " +
-                                result + " for expression " + expression);
-                    }
-                    return ((Boolean) result).booleanValue();
-                } catch (Exception t) {
-                    throw new IllegalArgumentException("Could not evaluate completion condition  " + expression, t);
-                }
+        private boolean evaluateCompletionCondition(ReturnValueEvaluator completeConditionEvaluator, Map<String, Object> tempVariables) {
+            try {
+                KogitoProcessContextImpl context = (KogitoProcessContextImpl) ContextFactory.fromNode(this);
```
Comment (pefernan) [inline jbpm/jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/ForEachNodeInstance.java:322]: Cast not needed

**Review by pefernan — APPROVED** [review]
Just a small comment, otherwhise +1

**Review by martinweiler — APPROVED** [review]
Great work @elguardian

**Comment by kie-ci3** [comment]
**PR job** `#5` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3554 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3554/5/display/redirect

**Test results:**
- PASSED: 3286
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3554/5/testReport/org.kie.kogito.quarkus.workflows/PojoServiceIT/testPojo/">org.kie.kogito.quarkus.workflows.PojoServiceIT.testPojo</a></summary>
<br/>Expected size: 1 but was: 2 in:<br/>{"age"=666, "name"="javieritoPerson"}
</details>

**Pull request merged**